File: csharpcore/StarterBot/Enums/BuildingType.cs
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace StarterBot.Enums
{
    public enum BuildingType
    {
        [JsonProperty("DEFENSE")]
        [EnumMember(Value = "DEFENSE")]
        Defense = 0,
        [JsonProperty("ATTACK")]
        [EnumMember(Value = "ATTACK")]
        Attack = 1,
        [JsonProperty("ENERGY")]
        [EnumMember(Value = "ENERGY")]
        Energy = 2,
        [JsonProperty("TESLA")]
        [EnumMember(Value = "TESLA")]
        Tesla = 3
    }
}

File: java/src/main/java/za/co/entelect/challenge/enums/BuildingType.java
package za.co.entelect.challenge.enums;

public enum BuildingType {
    DEFENSE(0),
    ATTACK(1),
    ENERGY(2);

    private int type;

    BuildingType(int type) {

        this.type = type;
    }

    public int getType() {
        return type;
    }

    public String buildCommand(int x, int y) {
        return String.format("%d,%d,%d", x, y, getType());
    }
}
